src/debug.cc

```diff
@@ -163,6 +164,28 @@ class DebugChannel
     virtual void log(const DebugMessageHeader &, const std::string &body) = 0;
 
 protected:
+    /// output iterator for DebugMessages
+    class Logger
+    {
+    public:
+        using difference_type = std::ptrdiff_t;
+        using value_type = DebugMessage;
+        using pointer = DebugMessage *;
+        using reference = DebugMessage &;
+        using iterator_category = std::output_iterator_tag;
+
+        Logger(DebugChannel &ch) : channel(ch) {}
+        Logger &operator=(const DebugMessage &message) {
+            if (Debug::Enabled(message.header.section, message.header.level))
+                channel.log(message.header, message.body);
+            return *this;
+        }
+        Logger &operator*() { return *this; }
```

Comment:
Just to clarify strange type (for the first operator) and no-op implementation (for all three operators):

```suggestion
        // These no-op operators are provided to satisfy LegacyOutputIterator requirements,
        // as is customary for similar STL output iterators like std::ostream_iterator.
        Logger &operator*() { return *this; }
```